### ISSUE TITLE
Unexpectedbalck root screen as well a 100% CPU load

### ISSUE THREAD

**Issue by bitstreamout** [yshui/compton#1150]
### Platform
OpenSUSE Slowroll 20231119

### GPU, drivers, and screen setup
Combination of NVIDIA GPU NVIDIA GeForce GTX 960M (GM107-A)  and Intel(R) HD Graphics 530 (SKL GT2)
where modesetting driver provides the modes used by nvidia driver

### Environment
lxsession with xfce4-panel and fvwm

### picom version
v10

<details>
<summary>Diagnostics</summary>

</details>

### Configuration:
<details>
    <summary>Configuration file</summary>

```**Version:** v10

### Extensions:

* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: Yes
* Config file used: /etc/xdg/picom.conf

### Drivers (inaccurate):

modesetting

### Backend: glx

* Driver vendors:
 * GLX: NVIDIA Corporation
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GTX 960M/PCIe/SSE2

### Backend: egl

* Driver vendors:
 * EGL: NVIDIA
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GTX 960M/PCIe/SSE2


```
</details>

### Steps of reproduction

1. start lxsession  via ~/.xsession with configured FVWM setup
2. start picom

### Expected behavior
No black root screen
### Current Behavior
Black root screen as well as partly invisible icons
### Stack trace

Missing auto detection of backend leads with default xrender to 100% load caused by Xorg server and as no one knows about this tool it has taken a while to determine the real cause !!!

### OpenGL trace

How to disable this tool for lxsession without uninstalling it??

### Other details

Nothing

**Comment by sndirsch** [comment]
Downstream Bugzilla ticket: https://bugzilla.suse.com/show_bug.cgi?id=1217786

**Comment by yshui** [comment]
this might have been solved by #984

**Comment by absolutelynothelix** [comment]
the upstream bug report mentions `xsetroot` that we don't and can't support for the same reason as `display` - they simply don't store the root background pixmap (aka wallpaper) in any of the corresponding atoms so there is no way to obtain it for us. this explains the black root screen.

the repeatedly printed error and, i guess, high cpu usage caused by this should be fixed by the #1001. also, this error is only present in the legacy xrender and iirc xr_glx_hybrid backends. you shouldn't use the legacy backends if you don't have a reason to and if you do you should report why.

try picom built from the latest commit in the next branch and report back if you still have issues.

closing as fixed in #1001.
